src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,81 @@
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onAdd: (val: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onAdd, selectedPost }) => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [name, setName] = useState('');
+  const [isErrorName, setIsErrorName] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [isErrorEmail, setIsErrorEmail] = useState(false);
+
+  const [comment, setComment] = useState('');
+  const [isErrorComment, setIsErrorComment] = useState(false);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setIsErrorName(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setIsErrorEmail(false);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment(e.target.value);
+    setIsErrorComment(false);
+  };
+
+  const handleReset = () => {
+    setComment('');
+    setEmail('');
+    setName('');
+    setIsErrorComment(false);
+    setIsErrorEmail(false);
+    setIsErrorName(false);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    setIsErrorName(!name.trim());
+    setIsErrorEmail(!email.trim());
+    setIsErrorComment(!comment.trim());
+
+    if (!name.trim() || !email.trim() || !comment.trim()) {
```

Comment:
Great that you're checking for empty inputs, but remember that you need to check not only for empty strings but also for strings that contain only spaces. You can use the `trim()` method before checking.